### DISCUSSION TITLE
Get MotionProperties with bodyInterface

### DISCUSSION THREAD

**Discussion by kozakromch** [jrouwe/JoltPhysics#1130] [Q&A] [answered]
Hello, thanks again for the excellent library.  In my task, I need to change the linear and angular damping for the bodies during some simulation cycles.
In some updates I need 0 damping , in some 1.
As far as I understand the interface, I have to use BodyId and BodyInterface to change parameters, but I can't directly get MotionPropery or set linear and angular damping. 

It seemed a little strange to me, because I can SetGravityFactor for the body through the BodyInterface, but can't change another parameters, and, as I understand, directly change the properties of motion using the Body* body is a bad practice.

So, is there a proper way to set MotionProperties? 
Thanks in advance)

**Comment by jrouwe** [comment, marked as answer]
The BodyInterface provides a lot of convenience functions to quickly set properties. I did not expose everything (mainly because I was lazy). To set these properties you would do something like:

```
BodyLockWrite lock(*physics_system->GetBodyLockInterface(), body_id);
if (lock.Succeeded())
{
	MotionProperties *mp = lock.GetBody().GetMotionProperties();
	mp->SetLinearDamping(linear_damping);
	mp->SetAngularDamping(angular_damping);
}
```

B.t.w. if you have a `Body *` then you can also set these properties directly, just make sure that it is not done while something else is reading the values (Jolt itself only reads this during very specific times in `PhysicsSystem::Update`).
